### PR TITLE
Fixes Organization not found error in LeftDrawerOrg

### LINKED ISSUE #1333: Organization not found error in left drawer
**Describe the bug**
When user profile is opened through the orglist page, an error is shown on the member page and redirected back to orglist page.

1. Login 
2. Click on the username on the bottom left

**Expected behavior**
No error should be shown on the member page and should not redirect back to orglist.

https://github.com/PalisadoesFoundation/talawa-admin/assets/123248648/be02ad60-17be-4c32-973a-a3bf5fe5b67e

### PR DESCRIPTION
**Pull request by nitishkumar333**
**What kind of change does this PR introduce?**

Bugfix

Fixes #1333 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/123248648/a009d4a4-9097-4e2a-923b-baa8925f6c34

**Does this PR introduce a breaking change?**
No

**Summary**

When user profile is opened through the orglist page, an error is shown on the member page and redirected back to orglist page. I fixed the issue by conditionally renderingthe MemberDetails component.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

File: src/components/LeftDrawer/LeftDrawer.module.css [hunk 1]
```diff
@@ -66,14 +66,14 @@
...
-  background-color: var(--bs-white);
   display: flex;
   align-items: center;
+  padding: 2.5rem 1rem 2.5rem 1rem;
 }
 
 .leftDrawer .profileContainer:focus {
   outline: none;
-  background-color: var(--bs-gray-100);
+  background-color: #5ac989;
```
Comment (rishav-jha-mech) [inline src/components/LeftDrawer/LeftDrawer.module.css:76]: Don't hard code the colours, use css variables

File: src/components/LeftDrawer/LeftDrawer.module.css [hunk 2]
```diff
@@ -96,6 +96,10 @@
   font-weight: 600;
 }
 
+.leftDrawer .profileContainer .profileText .primaryText .primaryTextWhite {
+  color: white;
```
Comment (rishav-jha-mech) [inline src/components/LeftDrawer/LeftDrawer.module.css:100]: use css variable here

File: src/components/LeftDrawer/LeftDrawer.module.css [hunk 3]
```diff
@@ -104,6 +108,14 @@
   text-transform: capitalize;
 }
 
+.leftDrawer .profileContainer .profileText .secondaryTextBlack {
+  color: var(--bs-secondary);
+}
+
+.leftDrawer .profileContainer .profileText .secondaryTextWhite {
+  color: rgb(230, 230, 230);
```
Comment (rishav-jha-mech) [inline src/components/LeftDrawer/LeftDrawer.module.css:116]: Here too

File: src/screens/MemberDetail/MemberDetail.tsx [hunk 4]
```diff
@@ -18,19 +19,29 @@ import Loader from 'components/Loader/Loader';
 
 type MemberDetailProps = {
   id: string; // This is the userId
+  from?: string;
+};
+type LocationStateProps = {
```
Comment (rishav-jha-mech) [inline src/screens/MemberDetail/MemberDetail.tsx:24]: You could have used MemberDetailProps here, why create LocationStateProps which have similar keys ?, this is redundant

File: src/components/LeftDrawer/LeftDrawer.module.css (hunk 1 above)
Comment (nitishkumar333) [reply src/components/LeftDrawer/LeftDrawer.module.css:76]: Sure, just noticed that the color is being taken care of by button variant so I will remove it

File: src/components/LeftDrawer/LeftDrawer.tsx [hunk 5]
```diff
@@ -137,33 +137,51 @@
...
+              data-testid="profileBtn"
+              variant={screenName === 'orglist' ? 'success' : 'light'}
+              className={styles.profileContainer}
+            >
+              <div className={styles.imageContainer}>
+                {userImage && userImage !== 'null' ? (
+                  <img src={userImage} alt={`profile picture`} />
+                ) : (
+                  <img
+                    src={`https://api.dicebear.com/5.x/initials/svg?seed=${firstName}%20${lastName}`}
```
[the remark this answers is not in the document]
Comment (palisadoes) [reply src/components/LeftDrawer/LeftDrawer.tsx:157]: Please fix this security vulnterability

File: src/components/LeftDrawerOrg/LeftDrawerOrg.tsx [hunk 6]
```diff
@@ -183,33 +183,51 @@
...
+              variant={screenName === 'orgdash' ? 'success' : 'light'}
+              className={styles.profileContainer}
+              data-testid="profileBtn"
+            >
+              <div className={styles.imageContainer}>
+                {userImage && userImage !== 'null' ? (
+                  <img src={userImage} alt={`profile picture`} />
+                ) : (
+                  <img
+                    src={`https://api.dicebear.com/5.x/initials/svg?seed=${firstName}%20${lastName}`}
```
[the remark this answers is not in the document]
Comment (palisadoes) [reply src/components/LeftDrawerOrg/LeftDrawerOrg.tsx:203]: Please fix this security vulnterability

**Comment by palisadoes** [comment]
Closing due to inactivity

**Comment by palisadoes** [comment]
1. Please fix the conflicting files.
2. We should get this PR merged
3. Let us know when you are done so the reviewers can look again.

**Comment by nitishkumar333** [comment]
@rishav-jha-mech removed the merge conflict, please review

File: src/components/LeftDrawer/LeftDrawer.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (nitishkumar333) [reply src/components/LeftDrawer/LeftDrawer.tsx:122]: @rishav-jha-mech  can you please guide me to fix this bug, I am not getting the cause of this issue

**Review by noman2002 — CHANGES_REQUESTED** [review]
Please fix the failing test.

**Comment by palisadoes** [comment]
@noman2002 The introspection tests will fail for a few days as the API has new functionality that is not reflected in Admin.

**Comment by palisadoes** [comment]
@nitishkumar333

1. This seems to have been fixed. I tried doing the steps in the video and they all work.
2. Please verify and close the PR if it has been addressed

**Comment by nitishkumar333** [comment]
@palisadoes  it's not fixed yet.

To reproduce the error, login to talawa-admin and directly click on leftDrawer username (without opening any other page)

Here is the video just after login.

https://github.com/PalisadoesFoundation/talawa-admin/assets/123248648/23cadfc0-552e-4a01-80dc-d2d3ec20622a

But, if we open any organization and then go back to homescreen then it will work because the URL doesn't get full changed (which it should be changed back to "/orglist" and not "/orglist/id=6776343783475434")
Here is the video

https://github.com/PalisadoesFoundation/talawa-admin/assets/123248648/20fd282d-1d99-4959-8562-cdecd6920607

**Comment by palisadoes** [comment]
1. Yes I see it now. 
2. Please fix the DOM vulnerability. It looks like you may have to escape some characters in the strings

**Comment by palisadoes** [comment]
Please fix the failing tests
- Code coverage is declining. The tests for the files you have submitted are not complete

**Review by noman2002 — CHANGES_REQUESTED** [review]
Please fix the DOM vulnerability.

**Comment by noman2002** [comment]
Closing due to inactivity.

**Pull request closed without merging**